core/common/src/main/java/alluxio/metrics/MetricsSystem.java

```diff
@@ -39,225 +44,251 @@
  * The syntax of the metrics configuration file is:
  * [instance].[sink|source].[name].[options]=[value]
  */
-@NotThreadSafe
-public class MetricsSystem {
+@ThreadSafe
+public final class MetricsSystem {
   private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
 
   // Supported special instance names.
   public static final String MASTER_INSTANCE = "master";
   public static final String WORKER_INSTANCE = "worker";
+  public static final String CLIENT_INSTANCE = "client";
+
+  public static final MetricRegistry METRIC_REGISTRY = initMetricRegistry();
```

Comment:
For the first comment, I meant if the method `initMetricRegistry` should just be in the static block of this class.

For the second comment, I was wondering if we could provide methods which only expose certain functionality of the metrics system. I notice there are some methods like `registerGaugeIfAbsent` which would benefit from this.